n2survey/plot/simple_comparison.py

```diff
@@ -0,0 +1,193 @@
+    for answer in combi[:, 0]:
+        answer_rows = np.where(combi[:, 0] == answer)
+        percentage[answer] = np.append(
+            np.reshape(combi[answer_rows, 1], newshape=(-1, 1)),
+            np.reshape(
+                np.round(
+                    no_of[answer_rows] / sum(no_of[answer_rows]) * 100, decimals=1
+                ),
+                newshape=(-1, 1),
+            ),
+            axis=1,
+        )
+    return percentage
+
+
+def form_x_and_y(df, totalbar=False, answer_supress=False):
+    """
+    This Function splits up the dataframe given to it to x and y components
+    it applies the get_percentages function to calculate Percentages from
+    the compared with Question.
+    It also filters out the answers specified by answer_supress.
+    If additional answers should be compared this function also splits them up
+    for the get_percentages function which only can handle one array, not
+    a list of arrays.
+    """
+    PERCENTAGES = []
+    for entry in df:
+        PERCENTAGE = get_percentages(entry, totalbar=totalbar)
+        totalbar = False
+        if answer_supress:
+            for answer in answer_supress:
+                if answer not in PERCENTAGE:
+                    print(f"{answer} does not exist for this question")
+                else:
+                    PERCENTAGE.pop(answer)
+        PERCENTAGES.append(PERCENTAGE.copy())
+
+    x = []
+    y = []
+    for PERCENTAGE in PERCENTAGES:
+        for entry in PERCENTAGE:
+            x.append(entry)
+            y.append(PERCENTAGE[entry])
+    return x, y
+
+
+def form_bar_positions(df, bar_positions=False, totalbar=False, answer_supress=False):
+    """
+    forms a complete list of bar positions for all bars, also the not
+    specified ones.
+    You can specify bar positions in a list, but if you
+    want to specify the last bar you have to specify every bar before that,
+    Maybe this can be improved for more user friendlieness.
+    --> maybe exchange to a bar distance dictionary?
+    if dictionary to complex for standard user, maybe list of tuples?
+    [('Total',1.5),('Yes', 0.2),] etc.
+    with (nameofbar, distance_from_previous_bar)
+    At the moment we calculate PERCENTAGE a second time in this function,
+    but the slowing of the whole plot function is neglectable due to this.
+    """
+    # start list of bar_positions
+    if bar_positions:
+        bar_positions_complete = bar_positions
+    else:
+        bar_positions_complete = [0]
+    COUNT = 0
+    BAR_POS_LENGTH = 0
+    for entry in df:
+        PERCENTAGE = get_percentages(entry, totalbar=totalbar)
+        if answer_supress:
+            for answer in answer_supress:
+                if answer not in PERCENTAGE:
+                    print(f"{answer} does not exist for this question")
+                else:
+                    PERCENTAGE.pop(answer)
+        BAR_POS_LENGTH = BAR_POS_LENGTH + len(PERCENTAGE)
+        totalbar = False
+        if all(
```

Comment:
Can't quite find the reference, but I believe it's better to connect multiple conditions with `and` rather than calling `all()` on a list, but I could be wrong...

Reply:
Definitively has more style, changed it.